### PR TITLE
Linux cleanup

### LINKED ISSUE #272: Create a proper abstraction for linux os strings
In handling all the Linux* streams I absentmindedly parsed everything as utf8 since I know that's ~basically true. But things like paths (which are a lot of the values we process) are technically just "arbitrary bytes", so we should have a proper LinuxOsString abstraction for the output and do the parsing/matching on byte-strings.

Right now nothing *bad* will happen if this assumption is broken. We'll just fail to parse whatever data we're reading and come back with default/empty values for the stream. Just a quality of implementation thing, and probably not one that will come up often (since non-utf8 paths are hard to make in the first place).

### PR DESCRIPTION
**Pull request by Gankra**
This adds a new strings submodule with LinuxOsString and LinuxOsStr types that provide
most of the functionality we need. Ultimately though minidump_processor feeds things that
truly are LinuxOsStrings through String::from_utf8_lossy because we need to do *something*
to output them to json/cli.

This also moves all the linux streams to properly using Cow and lifetimes.

Also includes a random bugfix I noticed while refactoring.

fixes #272

### CODE REVIEW

**Comment by Gankra** [comment]
New commits added:

* implemented my proposed new api for LinuxStreams (Fixes #273)
* pulled synth_minidump out to its own crate so other crates could depend on it as a dev_dependency and use it in their tests
* linux stream tests largely move to minidump_processor as only it actually parses them


I really didn't like the way we were handling these streams, the new refactor to have them be just key-value iterators seems way more principled and useful. This is a breaking change in that now minidump will not parse out any fields for LSB or CpuInfo, instead deferring the task to SystemInfo.

**Comment by Gankra** [comment]
Was this too much effort for a minimally important set of APIs? Maybe.

Was this bothering me too much to leave it be? Absolutely.

File: minidump-processor/Cargo.toml [hunk 1]
```diff
@@ -38,3 +38,4 @@ scroll = "0.10.2"
 
 [dev-dependencies]
 test-assembler = "0.1.5"
+synth-minidump = { path = "../synth-minidump" }
```
Comment (luser) [inline minidump-processor/Cargo.toml:41]: This is a good idea and I'm not sure why I never got around to doing it!

**Comment by luser** [comment]
I'm not going to approve this because that would imply that I reviewed it and I haven't, but it sounds good! Unless you explicitly request my review on things I work under the assumption that you will merge your own changes as needed.

File: minidump-processor/Cargo.toml (hunk 1 above)
Comment (Gankra) [reply minidump-processor/Cargo.toml:41]: I assumed you (reasonably) thought the subcrates could grab it as a pub symbol in test builds but sadly It Doesn't Work Like That/

**Comment by Gankra** [comment]
Yeah I'm pretty happy with this. Always appreciate just a high level "this makes sense" though.

**Pull request merged**